packages/lib-utils/src/types/k8s.ts

```diff
@@ -50,11 +64,64 @@ export type QueryParams = Partial<{
   labelSelector: string;
   fieldSelector: string;
   resourceVersion: string;
-  [key: string]: string;
+  [key: string]: string | number;
 }>;
 
 export type Patch = {
   op: string;
   path: string;
   value?: unknown;
 };
+
+export enum Operator {
+  Exists = 'Exists',
+  DoesNotExist = 'DoesNotExist',
+  In = 'In',
+  NotIn = 'NotIn',
+  Equals = 'Equals',
+  NotEqual = 'NotEqual',
+  GreaterThan = 'GreaterThan',
+  LessThan = 'LessThan',
+  NotEquals = 'NotEquals',
+}
+
+export type MatchExpression = {
+  key: string;
+  operator: Operator | string;
+  values?: string[];
+  value?: string;
+};
+
+export type MatchLabels = {
+  [key: string]: string;
+};
+
+export type Selector = Partial<{
+  matchLabels: MatchLabels;
+  matchExpressions: MatchExpression[];
+  [key: string]: unknown;
+}>;
+
+export type FilterValue = Partial<{
+  selected: string[];
+  all: string[];
+}>;
+
+export type GetGroupVersionKindForModel = (model: K8sModelCommon) => K8sResourceIdentifier;
+
+/**
+ * @deprecated Use K8sResourceIdentifier type instead. Support for type GroupVersionKind will be removed in a future release.
+ * @see K8sResourceIdentifier
+ * GroupVersionKind unambiguously identifies a kind.
+ * https://godoc.org/k8s.io/apimachinery/pkg/runtime/schema#GroupVersionKind
+ * TODO: Change this to a regex-type if it ever becomes a thing (https://github.com/Microsoft/TypeScript/issues/6579)
+ */
+export type GroupVersionKind = string;
+
+/**
+ * @deprecated Use GetGroupVersionKindForModel type instead. Support for type K8sResourceKindReference will be removed in a future release.
+ * @see GetGroupVersionKindForModel
+ * The canonical, unique identifier for a Kubernetes resource type.
+ * Maintains backwards-compatibility with references using the `kind` string field.
+ */
+export type K8sResourceKindReference = GroupVersionKind | string;
```

Comment:
I wish this would not be exported if we can avoid it... our goal is to get rid of it and I don't want to make a release with stuff we want to move away from.